tests/installation/yast2_firstboot.pm

```diff
@@ -0,0 +1,94 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Utilize OS using YaST2 Firstboot module
+# Doc: https://en.opensuse.org/YaST_Firstboot
+# Maintainer: Martin Loviska <mloviska@suse.com>
+
+use base "y2logsstep";
+use strict;
+use warnings;
+use testapi;
+use utils qw(zypper_call clear_console);
+use installation_user_settings qw(await_password_check enter_userinfo enter_rootinfo);
+use version_utils qw(is_sle is_opensuse);
+
+sub language_and_keyboard {
+    my $shortcuts = {
+        l => 'lang',
+        k => 'keyboard'
+    };
+    assert_screen 'lang_and_keyboard';
+    mouse_hide(1);
+    foreach (sort keys %${shortcuts}) {
+        send_key 'alt-' . $_;
+        assert_screen $shortcuts->{$_} . '_selected';
+        send_key_until_needlematch 'expanded_list', 'spc', 5, 7;
+        wait_screen_change(sub { send_key 'ret'; }, 5);
+    }
+    wait_screen_change(sub { send_key $cmd{next}; }, 7);
+}
+
+sub license {
+    my $self = shift;
+    return unless is_sle('>=12-sp4');
+    assert_screen('license-agreement');
+    $self->verify_license_has_to_be_accepted;
+    $self->accept_license;
+    wait_screen_change(sub { send_key $cmd{next}; }, 7);
+    # WA
+    send_key 'alt-a';
+    wait_screen_change(sub { send_key $cmd{next}; }, 7);
+    #
+}
+
+sub welcome {
```

Comment:
Initially I thought that we could reuse existing modules, at least we could do that for some. Have you also considered splitting these into separate test modules?

Reply:
yes, and there is not much to be reused from my point of view. For instance, *welcome.pm*. In the classical installer the module handles language and keyboard layout as well as product selection, whereas Firstboot module separates L&K view from Welcome view.

Reply:
My main concern is about this module being maintained separately and becoming monster. `welcome.pm` is mixing multiple purposes, but language and keyboard widgets are same I guess. Anyway, we can leave for now.